crates/mun_codegen/src/ir/body.rs

```diff
@@ -734,6 +734,7 @@ impl<'a, 'b, D: IrDatabase> BodyIrGenerator<'a, 'b, D> {
             ArithOp::Subtract => self.builder.build_int_sub(lhs, rhs, "sub"),
             ArithOp::Divide => self.builder.build_int_signed_div(lhs, rhs, "div"),
             ArithOp::Multiply => self.builder.build_int_mul(lhs, rhs, "mul"),
+            ArithOp::Remainder => self.builder.build_int_signed_rem(lhs, rhs, "rem"),
```

Comment:
I think this should differ for signed and unsigned integers: build_int_unsigned_rem

Reply:
probably the same needs to happen for div, now that I look at it